### ISSUE TITLE
runtime: go1.8 build fails when run as a Jenkins job (was #18551)

### ISSUE THREAD

**Issue by hartzell** [golang/go#19203]
This is a continuation of #18551.  I can now reproduce the problem on a smaller system and w/out involving [Spack](https://github.com/llnl/spack).

Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?

Using go-1.4 (built via Spack's [go-bootstrap package](https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/go-bootstrap/package.py)) to build v1.8.

### What operating system and processor architecture are you using (`go env`)?

```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH=""
GORACE=""
GOROOT="/isilon/Analysis/scratch/hartzelg/linuxbrew/Cellar/go/1.7.5/libexec"
GOTOOLDIR="/isilon/Analysis/scratch/hartzelg/linuxbrew/Cellar/go/1.7.5/libexec/pkg/tool/linux_amd64"
CC="/usr/bin/cc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build542396042=/tmp/go-build -gno-record-gcc-switches"
CXX="/usr/bin/c++"
CGO_ENABLED="1"
```

### What did you do?

I had been using Spack's [go package](https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/go/package.py) to build go and seeing the crashes reported in #18551.  I initially attributed it to the size of the machine and considered Spack a second candidate culprit.

I have replace Spack with this shell script (stop giggling...):

```bash
#!/bin/bash

set -x
set -e
set -u

cd /path/to/tmp
rm -rf poodle
mkdir poodle
cd poodle
wget https://storage.googleapis.com/golang/go1.8.src.tar.gz
tar -xzf go1.8.src.tar.gz
cd go/src
export GOROOT_BOOTSTRAP=/path/to/go-bootstrap-1.4
./all.bash
```

I can run this command from the command line successfully.

I have a Jenkins job set up with this snippet of Jobs DSL

```groovy
job('go-shell-script') {
  concurrentBuild(true)
  label(themachine')
  steps {
    shell '''#!/bin/bash
~me/build-go.sh
'''
  }
  publishers {
    mailer('me@work.com', false, true)
  }
}
```

The Jenkins master->slave connection is via SSH and it uses my account.

When I build it, I see the same symptoms I've been seeing in #18551.  The output is in [this gist](https://gist.github.com/hartzell/6d38d98e7d4ecec20fa499ac5f9cae58).

### What did you expect to see?

All tests passing in both command line and Jenkins build.

### What did you see instead?

See [the gist](https://gist.github.com/hartzell/6d38d98e7d4ecec20fa499ac5f9cae58) referenced above.

## Conclusions

It seems that running the build remotely via Jenkins is my new primary candidate.

**Comment by ALTree** [comment]
Another `runtime` test timeout: #18442
Another `runtime` test timeout: #19196 (also jenkins)

**Comment by hartzell** [comment]
@ALTree -- what convinces you that the Gentoo sandboxes have the same root problem as running via an ssh connection from Jenkins?

The Common Pitfalls section of the [SSH Slaves plugin page](https://wiki.jenkins-ci.org/display/JENKINS/SSH+Slaves+plugin) says that in the end the Jenkins job script ends up being run like:

>  [...] but OpenSSH runs this with "bash -c command ..." (or whatever your login shell is.)

#18442 seems to be fixated on the sandboxing restrictions and/but I don't think the shell invocation above is sandboxing anything, though I'm still trying to understand what *is* happening...

**Comment by ALTree** [comment]
> shell invocation above is sandboxing anything

My suspect was that jenkins was transparently sandboxing executions in a similar manner, but I'm absolutely not a jenkins expert so please disregard my comment if you're convinced it's not useful.

**Comment by ALTree** [comment]
Can you add a `t.Skip()` to `TestCrashDumpsAllThreads` in `go/src/runtime/crash_unix_test.go` and try to build?

**Comment by ALTree** [comment]
Yeah this is a dup of #19196. Closing since that one is older.

**Comment by hartzell** [comment]
@altree -- Testing as you requested.

The stack traces that I get don't always mention TestCrashDumpAllThreads, e.g. this one (from #18551):

https://gist.github.com/hartzell/232cf1a1624be49e278ea4f641eea01f

**Comment by hartzell** [comment]
Skipping that test gave me two successful builds in a row from Jenkins.

**Comment by ALTree** [comment]
The gist in #18551: https://gist.github.com/hartzell/89dbb8fe33c513b018de3dcedb87848a is stuck on `TestCrashDumpAllThreads` too.

https://gist.github.com/hartzell/89dbb8fe33c513b018de3dcedb87848a#file-spack-build-out-L106

 Why do you say it's not?

**Comment by ALTree** [comment]
OK leaving this open until we're sure that the underling issue is not the same as the other threads that I linked here.

**Comment by hartzell** [comment]
The [gist that does not mention TestCrashDumpAllThreads](https://gist.github.com/hartzell/232cf1a1624be49e278ea4f641eea01f) does mention TestGdbBacktrace, which seems to be involved in #18442.

**Comment by aclements** [comment]
@hartzell, are you still having this issue? If so, I'd like to figure out if the test is actually hanging or just being slow. Could you set the environment variable GO_TEST_TIMEOUT_SCALE=10 and see if you can still reproduce?

> The gist that does not mention TestCrashDumpAllThreads does mention TestGdbBacktrace, which seems to be involved in #18442.

It mentions TestGdbBacktrace, but TestGdbBacktrace isn't actually the running test. It's just blocked on testing-internal mechanisms waiting to be skipped.

**Comment by hartzell** [comment]
@aclements -- Repeating comment from another thread, I'm on vacation and miles/worlds away from work and access to this.  I'll see what I can do when I return in a couple of Mondays.

**Comment by bradfitz** [comment]
Punting to Go 1.10. This isn't a regression from Go 1.8 to Go 1.9 anyway.

**Comment by aclements** [comment]
@hartzell, no worries. Thanks in advance!

**Comment by hartzell** [comment]
@aclements -- I have access to this large machine again, but at the moment it is involved in a bug hunt.  I'll keep looking for an opportunity to recreate the symptoms and follow up.

**Comment by chlunde** [comment]
I think this can be closed as a duplicate of #19196, as this is also when spawning from a Java process.

**Comment by hartzell** [comment]
Closing it seems reasonable to me too.

@chlunde has a more amenable test case too.
